zookeeper-server/src/test/java/org/apache/zookeeper/common/ZKTrustManagerTest.java

```diff
@@ -148,9 +156,6 @@ public void testServerHostnameVerificationWithHostnameVerificationDisabled() thr
         X509Certificate[] certificateChain = createSelfSignedCertifcateChain(IP_ADDRESS, HOSTNAME);
         zkTrustManager.checkServerTrusted(certificateChain, null, mockSocket);
 
-        verify(mockInetAddress, times(0)).getHostAddress();
```

Comment:
These `verify` calls are used to assert for expected behavior of hostname verification in various use cases. Without the `verify` calls, we would lose some test coverage, and I think several of these tests would all be testing the same thing.

I'm not familiar with the new burningwave library. Do you know if there are any options it offers for intercepting these calls so that we could try to preserve this test coverage?

Reply:
I knew that we are going to lose these assertions, the problem is that the tests verified that these two methods were not called (or called N times).
We cannot spy the InetAddress object anymore.

we could create two "versions" of this test, if we are on JDK19+ we do it this way, and if we are on older JDKs we keep the mock and run the test the same way as before.
I don't think that there will be short term plans to abandon older JDKs soon (years)